### ISSUE TITLE
convert disease-variant-drug data into triples

### ISSUE THREAD

**Issue by nlwashington** [monarch-initiative/mckb#1]
here's a first pass at design of triples, given some of oscar's data.
Example1:
<pre>
indiv1 rdf:type glioma_of_the_brain
  rdfs:label “glioma of the brain with MGMT any promoter methylation”
  GENO:has_genotype gtindiv1
  monarch:responds_with_increased_benefit DrugBank:DB00853

DrugBank:DB00853 a owl:Class  #drugs will have to be curated.
DrugBank:DB00853 rdfs:label “temozolomide” 

gtindiv1 rdf:type GENO:methylation  #we might need to curate these kinds of genotypes
  RO:contained_in SO:promoter
  RO:targets NCBIGene:4255    #need to curate gene ids/fetch from NCBI
  refs:label “MGMT any promoter methylation”
</pre>
  
Example2:
<pre>
indiv2 rdf:type ductal_carcinoma
indiv2 rdfs:label “ductal carcinoma of the breast with ERBB2 G309A missense mutation”
indiv2 GENO:has_genotype gtindiv2

indiv2 monarch:reduced_sensitivity_toward CHEBI:49603      #lapatinib

gtindiv2 rdf:type GENO:sequence_alteration,SO:missense_variant
  rdfs:label “ERBB2 G309A missense mutation”
  GENO:is_sequence_variant_of NCBIGene:ERBB2
  faldo:location [ 
    a faldo:position,
    faldo:begin 390,  #preferable to use proper genomic location
    faldo:end 390 
    reference: protein_id]   #if we don’t map to the genomic level, could use the protein id here for relative position
  GENO:reference_nucleotide “X”  #not sure we have these relationships?
  GENO:altered_nucleotide “Y”
  GENO:results_in_amino_acid_change “A”

annotation180 a annotation
  hasSubject indiv2
  hasObject gtindiv2
  hasPredicate monarch:reduced_sensitivity_toward
  dc:source PMID:23220880
</pre>
has_genotype = GENO:0000222
and the genes/labels could be defined elsewhere, if desired.

basically, we create an individual that is the combination of the cancer/disease diagnosis with the given genotype.  this is what is related to the drug via the to-be-added response relationships (we can just put these in the monarch namespace for now).  @cmungall and @mbrush please help get the right relationships for the "genotype".  Example1 is a tough one. (note i didn't put in the reified annotation for example1, but it should be added.)

**Comment by cmungall** [comment]
Example 1, drug to disease phenotype part:

Following the more granular noctua style of modeling, we'd create an individual of the drug class.We could also use CHEBI here

```
indiv1 rdf:type glioma_of_the_brain
  rdfs:label “glioma of the brain with MGMT any promoter methylation”
  GENO:has_genotype gtindiv1
  monarch:responds_with_increased_benefit indiv2

indiv2 a CHEBI:72564    ## or a DrugBank:DB00853
## any other info about this particular instance of the drug
```

**Comment by kshefchek** [comment]
When attaching sources, what is the best way to model to relationship?  Would this technically require a 4-ary relationship? For example, (a drug X has affect on Disease Y when a patient has a genotype Z ) -> has source - PubMed:1234.

**Comment by kshefchek** [comment]
In regards, to querying information on variant/genotypes, I've added 4 queries/functions to g2p.py
There are 1106 genotypes linked to disease-therapy outcomes
1. _get_genotype_protein_info -> get 341 genotypes mapped to a protein_variant (but not to a cdna variant)
2. _get_genotype_cdna_info -> get 141 genotypes mapped to a protein_variant and cdna_variant (incomplete)
3. _get_fusion_copy_any_mutation_genotypes -> get 338 genotypes mapped to a gene in the variant_therapy table, but no protein_variant (and therefore cdna_variant).  Typically, this will capture fusion genes and copy/gain loss along with mutations labelled "any" (any mutation of gene X)
4. _get_genotypes_with_no_gene_protein_cdna_mapping -> get 286 genotypes not mapped to a  gene in therapy_variant.  This will capture rearrangement fusion genes, missense mutations where there are multiple mutations or the specific mutation is unknown, amplification copy number genes, and indels.

This will capture all of the genotypes and allow us to take different approaches for each case.  I'll start on cdna and protein mappings.

In regards to querying disease-variant-drug data, the query to get this information is in the function _get_disease_drug_genotype_relationship().

**Comment by nlwashington** [comment]
>When attaching sources, what is the best way to model to relationship? Would this technically require a 4-ary relationship? For example, (a drug X has affect on Disease Y when a patient has a genotype Z ) -> has source - PubMed:1234.

For this, I am thinking we could do one of a few things:
patient 
  has_genotype Z 
  has_phenotype Y 
  response_to_drug D 

with each of those having associations with the S,P,O.  in this case the "patient" is a proxy for "people".  

or we could make an instance of a disease class that is caused by a particular genotype/allele, and link that, like:
[disease instance caused by genotype Z]
  response_to_drug D

**Comment by kshefchek** [comment]
Both of those models make sense.  Perhaps modelling a patient is more scalable as we may eventually add more patient metadata that affects the disease/drug response.

**Comment by nlwashington** [comment]
the only caveat might be to model it as a population of patients, rather than a single one.  i think i've done something similar for populations of mice in IMPC.

**Comment by kshefchek** [comment]
For now I'll model it this way:
<pre>
indiv a GENO:0000110 (population)
        RO:0002200 (has_phenotype)  disease
        GENO:0000222 (has_genotype) genotype
        (response_to_drug) drug
</pre>

**Comment by kshefchek** [comment]
Transcripts are annotated in a table called transcript_priority as either "primary" or "secondary."  What sequence ontology terms can we map to these?  This is what I can find:
            'primary_transcript': 'SO:0000185',
            'transcript_secondary_structure_variant ': 'SO:0001596'

**Comment by nlwashington** [comment]
I am guessing that the transcript_priority the frequency of observance of a given transcript...  usually there is one transcript that is dominantly transcribed, and the one or more transcripts are present but at much less significant levels.  So this annotation is probably reflecting it's quantitative presence in the cell, as opposed to it's structure.  (the terms you selected above are more about the structure of the transcript, as opposed to the "order of importance or abundance")

**Comment by selewis** [comment]
Wouldn't they all be primary transcripts?

'transcript_secondary_structure_variant ': 'SO:0001596'

refers to the 2-dimensional structure of the transcript. Loops and such.

Priority to me would/might refer to whether this was the transcript that
the primers were designed from for doing the sequencing. (I'm out of
context here, so speculating). (my comment here is not relevant if this is
general RNAseq)

Otherwise I think Nicole is right and that these are referring to quantity.
-- Of course in one cell type which transcript is primary (most frequently
transcribed) could very well be reversed from what is primarily transcribed
in another cell type. And since there can be more than 2 isoforms we'd need
tertiary_transcript, quantenary_transcript, etc....

-S

**Comment by kshefchek** [comment]
This makes sense, for now I've added fake classes for these.  There are a few things here that I cannot find in SO or GENO.  I'll break these out into a new issue.

**Comment by kshefchek** [comment]
Closing since the above has been modelled into the current version.  Will break out changes/additions into new issues.
